### PR TITLE
Refactor - Remove hardcoded field names

### LINKED ISSUE #1019: Refactor - Remove hardcoded field names
### Descriptive summary

Occasionally, our GBL schema v1 field names bleed into the codebase. To be more consistent, and to prepare for the Aardvark metadata schema, we can leverage settings.yml to map Settings.FIELDS mappings to features and tests. 

### Expected behavior

Solr field names should be referenced from settings.yml in the codebase like:

```ruby
document[Settings.FIELDS.GEOM_TYPE]
```

### Actual behavior

Sometimes you see:

```ruby
document['layer_geom_type_s']
```

### Related work

Exploratory GBL Aardvark schema fork:
https://github.com/ewlarson/geoblacklight/tree/feature/aardvark-metadata-upgrade

### PR DESCRIPTION
**Pull request by ewlarson**
Updates settings.yml to include Settings.FIELDS values for all GBL v1 fields. Replaces hardcoded field names in application logic, views, and specs. Prepares GBL v3 for v4 Aardvark metadata schema updates.

Fixes #1019

```yaml
# Solr field mappings
FIELDS:
  :CREATOR: 'dc_creator_sm'
  :DESCRIPTION: 'dc_description_s'
  :FILE_FORMAT: 'dc_format_s'
  :GEOM_TYPE: 'layer_geom_type_s'
  :GEOMETRY: 'solr_geom'
  :IDENTIFIER: 'dc_identifier_s'
  :ISSUED: 'dct_issued_s'
  :LANGUAGE: 'dc_language_s'
  :LAYER_ID: 'layer_id_s'
  :LAYER_MODIFIED: 'layer_modified_dt'
  :PART_OF: 'dct_isPartOf_sm'
  :PUBLISHER: 'dc_publisher_s'
  :PROVENANCE: 'dct_provenance_s'
  :REFERENCES: 'dct_references_s'
  :RIGHTS: 'dc_rights_s'
  :SPATIAL_COVERAGE: 'dct_spatial_sm'
  :SUBJECT: 'dc_subject_sm'
  :SOURCE: 'dc_source_sm'
  :SUPPRESSED: 'suppressed_b'
  :TEMPORAL: 'dct_temporal_sm'
  :TITLE: 'dc_title_s'
  :TYPE: 'dc_type_s'
  :UNIQUE_KEY: 'layer_slug_s'
  :WXS_IDENTIFIER: 'layer_id_s'
  :YEAR: 'solr_year_i'
```

### CODE REVIEW

File: lib/generators/geoblacklight/templates/settings.yml [hunk 1]
```diff
@@ -21,24 +21,31 @@ OVERLAP_RATIO_BOOST: '2'
...
+  :DESCRIPTION: 'dc_description_s'
   :FILE_FORMAT: 'dc_format_s'
+  :GEOM_TYPE: 'layer_geom_type_s'
   :GEOMETRY: 'solr_geom'
-  :WXS_IDENTIFIER: 'layer_id_s'
-  :RIGHTS: 'dc_rights_s'
+  :IDENTIFIER: 'dc_identifier_s'
+  :ISSUED: 'dct_issued_s'
+  :LANGUAGE: 'dc_language_s'
+  :LAYER_ID: 'layer_id_s'
```
Comment (mejackreed) [inline lib/generators/geoblacklight/templates/settings.yml:32]: Is this redundant w/ `WXS_IDENTIFIER`?

**Review by mejackreed — COMMENTED** [review]
This is great! 🥳 

One question, should we introduce a pattern here for providing default values for newly introduced keys?

For example, I'm on GeoBlacklight v.3.x and I upgrade to the latest, `SUPRESSED` and `UNIQUE_KEY` are introduced, and my settings file doesn't have those.

I'm trying to remember how we have handled this in the past?

We could maybe do this in an initializer thing in `engine.rb`?

File: lib/generators/geoblacklight/templates/settings.yml (hunk 1 above)
Comment (ewlarson) [reply lib/generators/geoblacklight/templates/settings.yml:32]: Yeah, I like that idea for adding defaults. My LAYER_ID is a duplication of WXS_IDENTIFIER. The WXS_ prefix threw me. I wasn't sure what that intended (must be mappy).

Are you thinking something like...

```Ruby
module Geoblacklight
  class Engine < ::Rails::Engine
    Blacklight::Configuration.default_values[:view].split.partials = ['index']
    Blacklight::Configuration.default_values[:view].delete_field('list')
    
    # Settings.FIELDS Default Values
    Settings.FIELDS.IDENTIFIER = 'dc_identifier_s' unless Settings.FIELDS.IDENTIFIER.present?
    # ... A bunch more defaults here ...

    # GeoblacklightHelper is needed by all helpers, so we inject it
    # into action view base here.
    initializer 'geoblacklight.helpers' do
      ActionView::Base.send :include, GeoblacklightHelper
      ActionView::Base.send :include, CartoHelper
    end

    config.to_prepare do
      Geoblacklight.inject!
    end
  end
end
```

Can you access Settings in engine.rb?

**Comment by ewlarson** [comment]
New Settings.FIELDS key/vals added here:

```
:IDENTIFIER: 'dc_identifier_s'
:LANGUAGE: 'dc_language_s'
:LAYER_ID: 'layer_id_s'
:LAYER_MODIFIED: 'layer_modified_dt'
:SOURCE: 'dc_source_sm'
:SUPPRESSED: 'suppressed_b'
:TYPE: 'dc_type_s'
:UNIQUE_KEY: 'layer_slug_s'
```

File: lib/generators/geoblacklight/templates/settings.yml (hunk 1 above)
Comment (mejackreed) [reply lib/generators/geoblacklight/templates/settings.yml:32]: Yep something like that. I _think_ we could even:

```
    initializer 'config.initialize' do #referencing config gem
    #check if it exists and maybe set a default?
    # or even
    # Config.load_files('new_values_only.yml')
    end
```

Comment (ewlarson) [reply lib/generators/geoblacklight/templates/settings.yml:32]: Okok. Will attempt on Monday.

File: config/initializers/new_gbl_settings_defaults_3_3.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# New GeoBlacklight v3.3 Solr field mappings
+FIELDS:
+  :IDENTIFIER: 'dc_identifier_s'
+  :LANGUAGE: 'dc_language_s'
+  :LAYER_ID: 'layer_id_s'
```
Comment (eliotjordan) [inline config/initializers/new_gbl_settings_defaults_3_3.yml:5]: As @mejackreed suggested, this is redundant. This line can be removed and references to it changed to `WXS_IDENTIFIER`. Or vice versa, however I prefer WXS_IDENTIFIER as it's more descriptive.

**Review by eliotjordan — CHANGES_REQUESTED** [review]
This looks good! 👍 Just fix the WXS_IDENTIFIER / LAYER_ID redundancy and I think it's ready.

File: config/initializers/new_gbl_settings_defaults_3_3.yml (hunk 2 above)
Comment (ewlarson) [reply config/initializers/new_gbl_settings_defaults_3_3.yml:5]: Sure thing! Just curious, but what does the WXS_ mean? Is it pertinent to all potential identifiers?

**Comment by ewlarson** [comment]
Okay! Removed the redundant field. Rebased the PR. Anything left here?

**Review by eliotjordan — APPROVED** [review]

**Review by eliotjordan — APPROVED** [review]

**Pull request merged**